libbeat/idxmgmt/std.go

```diff
@@ -176,41 +213,55 @@ func (s *indexSupport) BuildSelector(cfg *common.Config) (outputs.IndexSelector,
 		st:    &s.st,
 	}, nil
 }
+func (m *indexManager) VerifySetup(loadTemplate, loadILM LoadMode) (bool, string) {
+	ilmComponent := newComponent(componentILM, m.support.enabled(componentILM), false, loadILM)
+	templateComponent := newComponent(componentTemplate, m.support.enabled(componentTemplate),
+		m.support.templateCfg.Overwrite, loadTemplate)
+
+	if ilmComponent.load && !templateComponent.load {
+		return false, "Loading ILM policy and write alias without loading template " +
+			"is not recommended. Check your configuration."
+	}
+	if templateComponent.load && !ilmComponent.load && ilmComponent.enabled {
+		return false, "Loading template with ILM settings whithout loading ILM policy and alias can lead " +
+			"to issues and is not recommended. Check your configuration"
+	}
+	var warn string
+	if !ilmComponent.load {
+		warn += "ILM policy and write alias loading not enabled. "
+	}
+	if !templateComponent.load {
+		warn += "Template loading not enabled."
+	}
+	return warn == "", warn
```

Comment:
Consider error as second return value.

Reply:
I didn't want to indicate it is an error, as the combinations are still valid, but we should print warnings.